### PR TITLE
chore: refactor datastore otherMethods: iOS & Android

### PR DESCRIPTION
**Pull request by wooj2**
Refactors documentation so that iOS and Android uses the same set of documentation with different code snippets as suggested by @richardmcclellan in this PR: https://github.com/aws-amplify/docs/pull/2729

I decided to choose the most recent wording updated by @jamesonwilliams , so the Android page will get those updates with this PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by richardmcclellan** [comment]
Can JS use the same common content as well?

File: docs/lib/datastore/fragments/android/other-methods/10_clear.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+Amplify.DataStore.clear(
+    { Log.i("MyAmplifyApp", "DataStore cleared") },
+    { error: DataStoreException? -> Log.e("MyAmplifyApp", "Error clearing DataStore", error) }
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/10_clear.md:18]: (I know this isn't in the spirit of the current PR, but while we're here...)
```suggestion
    { Log.e("MyAmplifyApp", "Error clearing DataStore", it) }
```

File: docs/lib/datastore/fragments/android/other-methods/20_start.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+);
+```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+Amplify.DataStore.start(
+    { Log.i("MyAmplifyApp", "DataStore started") },
+    { error: DataStoreException? -> Log.e("MyAmplifyApp", "Error starting DataStore", error) }
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/20_start.md:17]:
```suggestion
    { Log.e("MyAmplifyApp", "Error starting DataStore", it) }
```

File: docs/lib/datastore/fragments/android/other-methods/10_clear.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+Amplify.DataStore.clear(
+    () -> Log.i("MyAmplifyApp", "DataStore cleared"),
+    error -> Log.e("MyAmplifyApp", "Error clearing DataStore: ", error)
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/10_clear.md:7]:
```suggestion
    error -> Log.e("MyAmplifyApp", "Error clearing DataStore", error)
```

File: docs/lib/datastore/fragments/android/other-methods/10_clear.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+</amplify-block>
+<amplify-block name="RxJava">
+
+```java
+RxAmplify.DataStore.clear()
+    .subscribe(
+        () -> Log.i("MyAmplifyApp", "DataStore cleared"),
+        error -> Log.e("MyAmplifyApp", "Error clearing DataStore: ", error)
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/10_clear.md:30]:
```suggestion
        error -> Log.e("MyAmplifyApp", "Error clearing DataStore", error)
```

File: docs/lib/datastore/fragments/android/other-methods/20_start.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+Amplify.DataStore.start(
+    () -> Log.i("MyAmplifyApp", "DataStore started"),
+    error -> Log.e("MyAmplifyApp", "Error starting DataStore: ", error)
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/20_start.md:7]:
```suggestion
    error -> Log.e("MyAmplifyApp", "Error starting DataStore", error)
```

File: docs/lib/datastore/fragments/android/other-methods/20_start.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+</amplify-block>
+<amplify-block name="RxJava">
+
+```java
+RxAmplify.DataStore.start()
+    .subscribe(
+        () -> Log.i("MyAmplifyApp", "DataStore started"),
+        error -> Log.e("MyAmplifyApp", "Error starting DataStore: ", error)
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/20_start.md:28]:
```suggestion
        error -> Log.e("MyAmplifyApp", "Error starting DataStore", error)
```

File: docs/lib/datastore/fragments/android/other-methods/30_stop.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+Amplify.DataStore.stop(
+    () -> Log.i("MyAmplifyApp", "DataStore stopped"),
+    error -> Log.e("MyAmplifyApp", "Error stopped DataStore: ", error)
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/30_stop.md:7]:
```suggestion
    error -> Log.e("MyAmplifyApp", "Error stopped DataStore", error)
```

File: docs/lib/datastore/fragments/android/other-methods/30_stop.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+);
+```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+Amplify.DataStore.stop(
+    { Log.i("MyAmplifyApp", "DataStore stopped") },
+    { error: DataStoreException? -> Log.e("MyAmplifyApp", "Error stopping DataStore", error) }
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/30_stop.md:17]:
```suggestion
    { Log.e("MyAmplifyApp", "Error stopping DataStore", it) }
```

File: docs/lib/datastore/fragments/android/other-methods/30_stop.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+</amplify-block>
+<amplify-block name="RxJava">
+
+```java
+RxAmplify.DataStore.stop()
+    .subscribe(
+        () -> Log.i("MyAmplifyApp", "DataStore stopped"),
+        error -> Log.e("MyAmplifyApp", "Error stopping DataStore: ", error)
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/other-methods/30_stop.md:28]:
```suggestion
        error -> Log.e("MyAmplifyApp", "Error stopping DataStore", error)
```

File: docs/lib/datastore/fragments/ios/other-methods/10_clear.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+```swift
+Amplify.DataStore.clear { result in
+    switch result {
+    case .success:
+        print("DataStore cleared")
+    case .failure(let error):
+        print("Error clearing DataStore:  \(error)")
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/ios/other-methods/10_clear.md:7]:
```suggestion
        print("Error clearing DataStore: \(error)")
```

**Review by jamesonwilliams — APPROVED** [review]
Woo-hoo! This is great @wooj2, thanks.

File: docs/lib/datastore/fragments/native_common/other-methods.md [hunk 11]
```diff
@@ -23,28 +15,12 @@ If your app uses authentication, it is recommended to call `DataStore.clear()` o
 
 Synchronization starts automatically whenever you run any DataStore operation (`query()`, `save()`, `delete()`, `observe()`.) You can also explicitly begin the process with `DataStore.start()`:
```
Comment (wooj2) [inline docs/lib/datastore/fragments/native_common/other-methods.md:17]: Hmm just realized, on iOS, this needs to be publisher(), not observe.  Whoops!

**Comment by wooj2** [comment]
> Can JS use the same common content as well?

They don't have a code sample for stop, but once they do...we can make it all common :)

**Pull request merged**